guardian/transcription-service
# Support export to google doc

## What does this change?
The main aim here is to allow users to export a transcript to a google doc. This involves the following features:
Client:
 - A new 'Export' page. This is the first React page that we have that is not just index.html. The way Next.js works is that it creates a separate page.html file for each page. As such I have had to update the production handler for client side pages -previously we always just returned index.html
 - A new 'ExportButton' component to handle export, waiting for an export, and linking to the generated google doc
 - `export.ts` service on the client which handles authentication with google to get permsissions to write to google drive
 - I've refactored `authFetch` to allow the entire request object to be overridden

Server:
 - A new /export endpoint to trigger the export and handle all the various error cases by returning an error with an appropriate status code
 - `googleDrive.ts` - this takes the authentication token shared by the client and uses it to actually create files in google drive
 - A new 'getTranscriptionItem' function to read values from dynamodb
 - Modification to how we serve the client side app - as discussed in the first point above

## How to test
You should be able to deploy to CODE/PROD and then use the test event on the output handler lambda (change the userEmail field) to get an email sent to you with a working export link.

Alternatively locally, you can run `npm run output-handler::start` to get a dummy item into your dynamodb table which you can then use to test the export on https://transcribe.local.dev-gutools.co.uk/export?transcriptId=my-first-transcription


![export](https://freeagent-res.cloudinary.com/image/upload/c_limit,w_1200/dpr_auto,f_auto/website-images/netlify/blog__export-data__export-data-og.gif)
